### PR TITLE
Add pidfd_open() thread-group leader extension

### PR DESCRIPTION
**Pull request by brauner**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -680,6 +680,17 @@ point that out explicitly and clearly in the associated patches and Cc
   could carry `user.dbus` set to `1`, and Varlink sockets
   `user.varlink` set to `1` and so on.
 
+* Extend `pidfd_open()` to allow opening the thread-group leader based on the
+  id of an individual thread. Currently we do support:
```
Comment (keszybz) [inline README.md:684]: ID ?

File: README.md [hunk 2]
```diff
@@ -680,6 +680,17 @@ point that out explicitly and clearly in the associated patches and Cc
...
+
+  (1) `pidfd_open(1234, 0)` on a thread-group lader pid
```
Comment (keszybz) [inline README.md:686]: "leader PID"

File: README.md [hunk 3]
```diff
@@ -691,6 +691,13 @@ point that out explicitly and clearly in the associated patches and Cc
    **Use-Case:** Allow for a race free way to go from individual thread to
    thread-group leader pidfd.
 
+* Add a namespace ioctl to translate a pid between pid namespaces.
```
Comment (keszybz) [inline README.md:694]: PID ?

File: README.md [hunk 4]
```diff
@@ -691,6 +691,13 @@ point that out explicitly and clearly in the associated patches and Cc
...
+
+  **Use-Case:** This makes it possible to e.g., figure out what a given pid in
+  a pid namespace corresponds to in the callers pid namespace. For example, to
```
Comment (keszybz) [inline README.md:697]: "caller's"

File: README.md [hunk 5]
```diff
@@ -691,6 +691,13 @@ point that out explicitly and clearly in the associated patches and Cc
...
+  figure out what the pid of pid 1 inside of the pid namespace is in the
+  ancestors pid namespace.
```
Comment (keszybz) [inline README.md:699]: "ancestor's"

**Review by keszybz — CHANGES_REQUESTED** [review]

File: README.md [hunk 6]
```diff
@@ -680,6 +680,23 @@ point that out explicitly and clearly in the associated patches and Cc
   could carry `user.dbus` set to `1`, and Varlink sockets
   `user.varlink` set to `1` and so on.
 
+* Extend `pidfd_open()` to allow opening the thread-group leader based on the
+  PID of an individual thread. Currently we do support:
+
+  (1) `pidfd_open(1234, 0)` on a thread-group lader PID
```
Comment (keszybz) [inline README.md:686]: Still "lader"

File: README.md [hunk 7]
```diff
@@ -680,6 +680,23 @@ point that out explicitly and clearly in the associated patches and Cc
...
+
+   Add an option to go from individual thread to thread-group leader.
+
+   **Use-Case:** Allow for a race free way to go from individual thread to
+   thread-group leader pidfd.
+
+* Add a namespace ioctl to translate a PID between PID namespaces.
+
+  **Use-Case:** This makes it possible to e.g., figure out what a given PID in
+  a pid namespace corresponds to in the caller's PID namespace. For example, to
```
Comment (keszybz) [inline README.md:697]: "pid namespace" vs. "PID namespace". Either way is fine, but please be consistent.

File: README.md [hunk 8]
```diff
@@ -680,6 +680,23 @@ point that out explicitly and clearly in the associated patches and Cc
...
+  figure out the PID of PID 1 inside of a given PID namespace is.
```
Comment (keszybz) [inline README.md:698]: "what" is missing here.

**Review by keszybz — CHANGES_REQUESTED** [review]

File: README.md (hunk 6 above)
Comment (brauner) [reply README.md:686]: ups

**Comment by keszybz** [comment]
This time I better merge it without looking ;)

**Pull request merged**